### DISCUSSION TITLE
How to implement PIXI.Loader's defaultQueryString for PIXI.Assets?

### DISCUSSION THREAD

**Discussion by Giacom** [pixijs/pixijs#8900] [Q&A] [answered]
PIXI.Loader had a way to automatically [append a query string](https://pixijs.download/v6.1.1/docs/PIXI.Loader.html#defaultQueryString) to the end of the URL.

I cannot find any way to do this for PIXI.Assets, any suggestions?

**Comment by bigtimebuddy** [comment]
I don't think this feature made it over. Could be a good feature addition to Assets?

**Reply by Giacom** [reply]
That's a shame, don't suppose there's some workaround I can do in the meantime?

**Reply by bigtimebuddy** [reply]
Why not manually add a query string parameter to your URLs?

**Reply by Giacom** [reply]
It's a lot of URLs that I hoped I wouldn't have to clutter up my manifest of assets with by adding them. 

Is there perhaps a way to do with the PIXI extensions, something like adding middleware in the PIXI.Loader.

**Reply by bigtimebuddy** [reply]
Here's an option to replace the `loadTextures` extension with one that appends a query string to URL. 
https://codesandbox.io/s/pixijs-examples-assets-resolver-v2336d

This approach is limited because it will only work with image textures. However, you could extend this to other types as well. This really just needs to be a feature.

**Reply by Giacom** [reply]
Oh thanks for the example. 

Yeah I hope it could just be added as an init option, would be very handy!

**Reply by Giacom** [reply]
I tried adding a param to the end of all my URLs manually but loadBitmapFont uses path.extname which makes the check for the extension incorrect and causes it to fail to parse.

**Reply by bigtimebuddy** [reply]
That seems like a bug. Can you file an issue with an example?

**Reply by Giacom** [reply]
Sure thing.

**Comment by Giacom** [comment, marked as answer]
An defaultSearchParams Assets init option was added in v7.1.0
